internal/resources/slo/data_source_slo.go

```diff
@@ -0,0 +1,366 @@
+package slo
+
+import (
+	"context"
+
+	gapi "github.com/grafana/grafana-api-golang-client"
+	"github.com/grafana/terraform-provider-grafana/internal/common"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func DatasourceSlo() *schema.Resource {
+	return &schema.Resource{
+		Description: `
+Datasource for retrieving all SLOs.
+		
+* [Official documentation](https://grafana.com/docs/grafana-cloud/slo/)
+* [API documentation](https://grafana.com/docs/grafana-cloud/slo/api/)
+				`,
+		ReadContext: datasourceSloRead,
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+		Schema: map[string]*schema.Schema{
+			"slos": &schema.Schema{
+				Type:        schema.TypeList,
+				Computed:    true,
+				Description: `Returns a list of all SLOs"`,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
```

Comment:
Did you try using the `common.CloneResourceSchemaForDatasource` function to build this schema? Normally, we re-use the schema from the resource. Easier to maintain